torch/csrc/autograd/custom_function.h

```diff
@@ -5,11 +5,41 @@
 
 namespace torch { namespace autograd {
 
-TORCH_API std::vector<Variable> _wrap_outputs(
-  const std::unordered_set<at::TensorImpl*> &inputs,
+TORCH_API variable_list _wrap_outputs(
+  const variable_list &input_vars,
   const std::unordered_set<at::TensorImpl*> &non_differentiable,
   const std::unordered_set<at::TensorImpl*> &dirty_inputs,
   const at::ArrayRef<Variable> raw_outputs,
   const std::shared_ptr<Function> &cdata);
 
+// To use custom autograd operations implement a CFunction subclass.
+// class MyFunction : public CFunction {
+//   static variable_list forward(AutogradContext *ctx, variable_list inputs);
+//
+//   static variable_list backward(AutogradContext *ctx, variable_list grad_output);
+// };
+// To use MyFunction
+// MyFunction::apply(inputs)
+
+template <class T>
+struct CFunction {
+  static void apply(variable_list&& inputs);
```

Comment:
You're probably going to want a non rvalue-reference version of this overload, for cases when people don't want to move their variable list into apply (similar to how Function is setup)